drivers/r2dbc-driver/src/main/kotlin/app/cash/sqldelight/driver/r2dbc/R2dbcDriver.kt

```diff
@@ -13,15 +13,47 @@ import kotlinx.coroutines.reactive.asFlow
 import kotlinx.coroutines.reactive.awaitFirstOrNull
 import kotlinx.coroutines.reactive.awaitSingle
 
-class R2dbcDriver(private val connection: Connection) : SqlDriver {
+class R2dbcDriver(private val connection: Connection, private val replaceParameter: Boolean) : SqlDriver {
```

Comment:
Yeah, using enums would be better.

Reply:
I agree this would increase readability. I will change it (probably not before next week). I also thought about deriving the correct value during runtime, but had no idea how this could be done. I couldn't find the information which dialect was used during compile time in the runtime code. That's the only reason I added this parameter in the first place.

Reply:
I _think_ we could add all of the R2DBC drivers we'd want to support as `compileOnly` dependencies on our R2DBC driver, and then do some check like this:
```kotlin
if (connection is PostgresR2DBCConnection) {
 separator = INDEXED
}
```
but I'm not 100% sure if it's as easy as that.